eva-server/src/main/java/uk/ac/ebi/eva/server/ws/contigalias/ContigAliasService.java

```diff
@@ -0,0 +1,171 @@
+            if (translatedContig.equals("")) {
+                variantsListAfterTranslatedContig.add(variant);
+            } else {
+                variantsListAfterTranslatedContig.add(createVariantsWithNewContig(variant, translatedContig));
+            }
+        }
+        return variantsListAfterTranslatedContig;
+    }
+
+    public VariantWithSamplesAndAnnotation createVariantsWithNewContig(VariantWithSamplesAndAnnotation variant,
+                                                                       String newContig) {
+        VariantWithSamplesAndAnnotation variantWithNewContig = new VariantWithSamplesAndAnnotation(newContig,
+                variant.getStart(), variant.getEnd(), variant.getReference(), variant.getAlternate(),
+                variant.getMainId());
+        variantWithNewContig.setAnnotation(variant.getAnnotation());
+        variantWithNewContig.addSourceEntries(variant.getSourceEntries());
+
+        return variantWithNewContig;
+    }
+
+    public List<FeatureCoordinates> getFeatureCoordinatesWithTranslatedContig(List<FeatureCoordinates> featuresList,
+                                                          ContigNamingConvention contigNamingConvention) {
+        if (skipContigTranslation(contigNamingConvention)) {
+            return featuresList;
+        }
+
+        List<FeatureCoordinates> featureListTranslatedContig = new ArrayList<>();
+        for (FeatureCoordinates feature : featuresList) {
+            String translatedContig = translateContigFromInsdc(feature.getChromosome(), contigNamingConvention);
+            if (translatedContig.equals("")) {
+                featureListTranslatedContig.add(feature);
+            } else {
+                featureListTranslatedContig.add(new FeatureCoordinates(feature.getId(), feature.getName(),
+                        feature.getFeature(), translatedContig, feature.getStart(), feature.getEnd()));
+            }
+        }
+
+        return featureListTranslatedContig;
+    }
+
+    public Annotation getAnnotationWithTranslatedContig(Annotation annotation,
+                                                        ContigNamingConvention contigNamingConvention) {
+        String translatedContig = translateContigFromInsdc(annotation.getChromosome(), contigNamingConvention);
+        if (translatedContig.isEmpty()) {
+            return annotation;
+        } else {
+            return new Annotation(translatedContig, annotation.getStart(), annotation.getEnd(),
+                    annotation.getVepVersion(), annotation.getVepCacheVersion(), annotation.getXrefs(),
+                    annotation.getConsequenceTypes());
+        }
+    }
+
+    public String translateContigFromInsdc(String genbankContig, ContigNamingConvention contigNamingConvention) {
+        if (contigNamingConvention == null) {
+            return "";
+        }
+        String url = contigAliasUrl + CONTIG_ALIAS_CHROMOSOMES_GENBANK_ENDPOINT + genbankContig;
```

Comment:
I know the endpoint is still called genbank but we should change it.
This is the INSDC accession and genbank can also issue a contig name that might be different from ENA's 

```suggestion
    public String translateContigFromInsdc(String insdcContig, ContigNamingConvention contigNamingConvention) {
        if (contigNamingConvention == null) {
            return "";
        }
        String url = contigAliasUrl + CONTIG_ALIAS_CHROMOSOMES_GENBANK_ENDPOINT + insdcContig;
```